### PR TITLE
Formatting to mongo query

### PR DESCRIPTION
**Pull request by CodMonk**
Added support for formatting to mongo db queries

### CODE REVIEW

**Comment by jakeboone02** [comment]
Hey, @CodMonk!  Thanks for this PR.  I hope it's useful.  Before I merge it, could you add some tests and update the README.md file?  Also check out the failing build (looks like a simple "Unnecessary semicolon" error in the lint step).

**Comment by CodMonk** [comment]
Hey @jakeboone02 
Updated the source. this should be fixed now.

**Comment by jakeboone02** [comment]
Alright, thanks for the updates @CodMonk.  A few more tweaks and we should be ready:
- I'd like to rename the query format identifier to `"mongodb"` since that's the official name, not just "mongo".
- We need to maintain 100% code coverage. (I know it's a meaningless metric, but it's sort of a point of pride now.) The codecov bot puts comments in the ["Files changed" tab](https://github.com/react-querybuilder/react-querybuilder/pull/214/files) for lines that aren't covered by tests.

And a couple of questions:
- Is the trailing comma in the resulting query actually valid MongoDB code? Even if it is, I think I'd prefer it if there were no trailing comma.
- Just curious, but why is the "not" operator not supported? Does MongoDB not have a corresponding keyword or query format?

**Comment by CodMonk** [comment]
Thank you for feedback @jakeboone02 . I will correct the test case and make the format to be `mongodb`. 

- Trailing comma was due to recursive function's last execution, I will handle that case.
- mongodb supports not operator but I did not find a way to do something like `not (expr1 ,expr 2 ... expr n)` . I will see if we can handle that case also.

**Comment by CodMonk** [comment]
MongoDB currently has open ticket for using NOT as top-level operator which is not solved yet, This can be fixed by using NOR operator instead of NOT but this might lead to confusion for some of the end-users.
Let me know what you think @jakeboone02  ?

Refer:
https://jira.mongodb.org/browse/SERVER-1454
https://stackoverflow.com/questions/25270396/mongodb-how-to-invert-query-with-not

File: README.md [hunk 1]
```diff
@@ -553,6 +553,7 @@ const query = {
 
 console.log(formatQuery(query, 'sql')); // '(firstName = "Steve" and lastName = "Vai")'
 console.log(formatQuery(query, 'parameterized')); // { sql: "(firstName = ? and lastName = ?)", params: ["Steve", "Vai"] }
+console.log(formatQuery(query, 'mongodb')); // {$and:[{firstName:{$eq:"Steve"}},{lastName:{$eq:"Vai"}},]}
```
Comment (jakeboone02) [inline README.md]: Small issue: there's still a trailing comma in the `mongodb` example.

**Comment by jakeboone02** [comment]
@CodMonk - It's up to you whether you want to proceed with using `$nor` or wait for `$not` to be implemented.  I don't really have an opinion either way.

**Comment by CodMonk** [comment]
Hi @jakeboone02 
trailing comma has been removed from example now. I want to wait sometime for `$not` to be implement as top-level operator.

**Comment by jakeboone02** [comment]
@all-contributors please add @CodMonk for code, doc, and test.

**Pull request merged**